### PR TITLE
fix(#1508): audit quality sweep — path sep, null-check, timeout, docs, tests

### PR DESCRIPTION
**Pull request by nogeenharrie**
## Summary

Fixes five code-quality issues identified in #1508:

- **`src/util.cpp` — `normalizeFolderPath`**: Remove forbidden `QDir::separator()` and `QDir::toNativeSeparators()` calls (CLAUDE.md prohibits `QDir::separator()` in path checks). The function now always appends `'/'`, matching Qt's internal path convention. Callers needing native separators can convert themselves.

- **`src/configdialog.cpp` — `validate()`**: Add null-check before dereferencing `QTableWidgetItem*` returned by `QTableWidget::item()`. The Qt docs state `item()` returns `nullptr` for cells never explicitly set; the previous code would crash unconditionally on such cells.

- **`src/configdialog.cpp` — `isQrencodeAvailable()`**: Pass `2000` ms to `waitForFinished()` instead of the infinite-timeout default, preventing the UI from hanging indefinitely on stalled network filesystems.

- **`src/pass.h` — `setEnvVar()` doc**: Expand the doc-block to make the trailing-`=` contract unambiguous. The existing implementation asserts and warns if `'='` is missing; the prior one-liner `@param` description was easy to miss.

- **`tests/auto/util/tst_util.cpp`**: Update existing `normalizeFolderPath` tests to match the new forward-slash contract; add `normalizeFolder()` covering idempotency, append-on-missing-slash, empty-string → `"/"`, and backslash-terminated input.

## Test plan

- [x] `make -j4` — zero errors
- [x] `QT_QPA_PLATFORM=offscreen ./tests/auto/util/tst_util` — 139 passed, 0 failed
- [x] `QT_QPA_PLATFORM=offscreen ./tests/auto/settings/tst_settings` — 117 passed, 0 failed
- [ ] CI green on all platforms

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Bug Fixes**
  * Prevented crashes during configuration validation when some profile rows are partially populated.
  * Fixed potential indefinite waiting when checking qrencode availability by adding a bounded timeout.

* **Documentation**
  * Clarified environment variable key requirements for setting/removing variables.
  * Updated folder-path documentation to guarantee forward slashes and a trailing `/`.

* **Tests**
  * Strengthened unit tests to enforce the updated `normalizeFolderPath()` contract, including edge cases and idempotency.

### CODE REVIEW

File: src/configdialog.cpp [hunk 1]
```diff
@@ -897,7 +899,7 @@ auto ConfigDialog::isQrencodeAvailable() -> bool {
 #else
   QProcess which;
   which.start("which", QStringList() << "qrencode");
-  which.waitForFinished();
+  which.waitForFinished(2000);
   QtPassSettings::setQrencodeExecutable(
       which.readAllStandardOutput().trimmed());
   return which.exitCode() == 0;
```
[the remark this answers is not in the document]
Comment (nogeenharrie) [reply src/configdialog.cpp:909]: Fixed — if `waitForFinished(2000)` times out the process is now killed before returning `false`, preventing partial output from being stored as the executable path.

[the remark this answers is not in the document]
Comment (nogeenharrie) [reply src/configdialog.cpp]: Same fix — process killed on timeout, returning false immediately.

File: tests/auto/util/tst_util.cpp [hunk 2]
```diff
@@ -310,41 +312,30 @@ void tst_util::cleanupTestCase() {
...
-            QString("Expected 'test\\\\subdir\\\\', got '%1'").arg(result)));
-  }
+  QVERIFY(result.endsWith('/'));
+  QCOMPARE(result, QStringLiteral("test/"));
 
-  // Mixed separators test
+  // Nested path without trailing slash.
+  result = Util::normalizeFolderPath("test/subdir");
+  QVERIFY(result.endsWith('/'));
+  QCOMPARE(result, QStringLiteral("test/subdir/"));
```
[the remark this answers is not in the document]
Comment (nogeenharrie) [reply tests/auto/util/tst_util.cpp:332]: `QVERIFY` is idiomatic Qt Test for simple boolean assertions and the project's existing tests use it consistently. Not changing to `QVERIFY2` without a project-wide decision.
